microbench/src/main/java/io/netty/microbenchmark/common/IsValidIpV6Benchmark.java

```diff
@@ -0,0 +1,306 @@
+        if (colonsAtStart) {
+            // FIX "IP6 mechanism syntax #ip6-bad1"
+            // An IPV6 address cannot start with a single ":".
+            // Either it can start with "::" or with a number.
+            if (ip.charAt(curIdx + 1) != ':') {
+                return false;
+            }
+            doubleColon = true;
+            prevColon = true;
+            colons = 2;
+            curIdx += 2;
+        }
+
+        int dots = 0;
+        int wordStart = curIdx;
+        int wordLen = 0;
+        loop: for (; curIdx <= lastIdx; curIdx++) {
+            c = ip.charAt(curIdx);
+            switch (c) {
+            // case for the last 32-bits represented as IPv4 x:x:x:x:x:x:d.d.d.d
+            case '.':
+                if (dots == 3 || !isValidIpV4Word_new(ip, wordStart, wordStart + wordLen)) {
+                    return false;
+                }
+                dots++;
+                wordStart = curIdx + 1;
+                wordLen = 0;
+                break;
+            case ':':
+                if (colons == 7 || dots > 0) {
+                    return false;
+                }
+                if (prevColon) {
+                    if (doubleColon) {
+                        return false;
+                    }
+                    doubleColon = true;
+                } else {
+                    prevColon = true;
+                }
+                colons++;
+                wordStart = curIdx + 1;
+                wordLen = 0;
+                break;
+            case '%':
+                // Strip the interface name/index after the percent sign
+                break loop;
+            default:
+                if (wordLen > 3 || !isValidHexChar(c)) {
+                    return false;
+                }
+                wordLen++;
+                prevColon = false;
+            }
+        }
+
+        // Check if we have an IPv4 ending
+        if (dots > 0) {
+            return dots == 3 &&
+                   // normal case (6 single colons) or a special case with double colon
+                   // such as ::1:2:3:4:5:d.d.d.d or ::ffff:5.6.7.8 allows to 7 colons with an IPv4 ending
+                   ((colons == 6 && !doubleColon) || (colons <= 7 && colonsAtStart)) &&
+                   // validate last word
+                   isValidIpV4Word_new(ip, wordStart, wordStart + wordLen);
+        }
+
+        // If we're at then end and we haven't had 7 colons then there is a
+        // problem unless we encountered a doubleColon
+        return (colons == 7 || doubleColon) &&
+               // If we have an empty word at the end, it means we ended in either a : or a .
+               // If we did not end in :: then this is invalid
+               (wordLen > 0 || (prevColon && doubleColon));
+    }
+
+    // Tests
+
+    @Benchmark
+    public boolean isValidIpV6Address_old() {
+        return isValidIpV6Address_old(ip);
+    }
+
+    @Benchmark
+    public boolean isValidIpV6Address_new() {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this method name to match the regular expression '^[_a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS00100)